internal/handlers/authorize/post.go

```diff
@@ -0,0 +1,81 @@
+package authorize
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"github.com/kamencov/go-musthave-diploma-tpl/internal/customerrors"
+	"github.com/kamencov/go-musthave-diploma-tpl/internal/logger"
+	"github.com/kamencov/go-musthave-diploma-tpl/internal/service/auth"
+	"net/http"
+)
+
+type Handler struct {
+	ctx         context.Context
+	authService *auth.ServiceAuth
+	log         *logger.Logger
+}
+
+func NewHandler(ctx context.Context, authService *auth.ServiceAuth, log *logger.Logger) *Handler {
+	return &Handler{
+		ctx:         ctx,
+		authService: authService,
+		log:         log,
+	}
+}
+
+func (h *Handler) Post(w http.ResponseWriter, r *http.Request) {
+	body := RequestBody{}
+
+	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
+		h.log.Error("error authorize", "error:", err)
+		apiError, _ := json.Marshal(customerrors.APIError{Message: "incorrect body"})
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write(apiError)
+		return
+	}
+
+	token, err := h.authService.AuthUser(h.ctx, body.Login, body.Password)
+	if err != nil {
+		if errors.Is(err, customerrors.ErrNotFound) {
+			h.log.Error("error authorize", "error:", err)
+			apiError, _ := json.Marshal(customerrors.APIError{Message: "cannot find user"})
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write(apiError)
+			return
+		}
+
+		if errors.Is(err, customerrors.ErrIsTruePassword) {
```

Comment:
Код ошибки немного вводит в заблуждение. 

```suggestion
		if errors.Is(err, customerrors.ErrWrongPassword) {
```